packages/trader/src/Modules/Trading/Components/Form/ContractType/contract-type-widget.jsx

```diff
@@ -5,107 +5,95 @@ import { localize } from '@deriv/translations';
-    };
-
-    componentDidMount() {
-        document.addEventListener('mousedown', this.handleClickOutside);
-    }
-
-    componentWillUnmount() {
-        document.removeEventListener('mousedown', this.handleClickOutside);
-    }
-
-    componentDidUpdate() {
-        if (this.props.list.length !== this.state.list.length && !this.state.on_input_change) {
-            // reset internal list state when contracts list is changed in store (on contracts_for response)
-            this.setState({ list: this.props.list });
+const ContractTypeWidget = ({ is_equal, name, value, list, onChange }) => {
+    const wrapper_ref = React.useRef(null);
+    const [is_dialog_open, setDialogVisibility] = React.useState(false);
+    const [is_info_dialog_open, setInfoDialogVisibility] = React.useState(false);
+    const [localList, setLocalList] = React.useState(list);
+    const [selected_category, setSelectedCategory] = React.useState(null);
+    const [search_query, setSearchQuery] = React.useState('');
+    const [item, setItem] = React.useState(null);
+
+    React.useEffect(() => {
+        document.addEventListener('mousedown', handleClickOutside);
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, []);
+
+    React.useEffect(() => {
+        if (list.length !== localList.length) {
+            setLocalList(list);
         }
-    }
+    }, [list]);
 
-    handleCategoryClick = ({ label }) => this.setState({ selected_category: label });
+    const handleCategoryClick = ({ label }) => setSelectedCategory(label);
 
-    handleSelect = (item, e) => {
+    const handleSelect = (selectedItem, e) => {
         if (e.target.id !== 'info-icon') {
-            this.setState(
-                {
-                    is_dialog_open: false,
-                    is_info_dialog_open: false,
-                    item,
-                },
-                () => {
-                    if (item.value !== this.props.value) {
-                        this.props.onChange({ target: { name: this.props.name, value: item.value } });
-                    }
-                }
-            );
+            setDialogVisibility(false);
+            setInfoDialogVisibility(false);
+            setItem(selectedItem);
         }
     };
 
-    handleInfoClick = item => {
-        this.setState(state => ({
-            is_info_dialog_open: !state.is_info_dialog_open,
-            item,
-        }));
-    };
+    React.useEffect(() => {
+        if (item && item.value !== value) {
+            onChange({ target: { name, value: item.value } });
+        }
+    }, [item]);
+
+    const handleInfoClick = ClickedItem => {
+        setInfoDialogVisibility(!is_info_dialog_open);
 
-    handleNavigationClick = item => {
-        this.setState({ item });
+        setItem(ClickedItem);
     };
 
-    setWrapperRef = node => {
-        this.wrapper_ref = node;
+    const handleNavigationClick = naveClickedItem => {
```

Comment:
what is `naveClickedItem`?

Reply:
when clicking on info icon of each contract item, we can see a carousel on the bottom which selected item for that is based on the item you select(ex: Rise/Fall)
![image](https://user-images.githubusercontent.com/82078941/146002997-ff4e1e5c-6b74-4b9f-a5d3-da719316fb64.png)